pkg/reconciler/tenancy/workspacemounts/workspacemounts_controller.go

```diff
@@ -197,20 +197,65 @@ func (c *Controller) process(ctx context.Context, key string) (bool, error) {
 	logger := logging.WithObject(klog.FromContext(ctx), workspace)
 	ctx = klog.NewContext(ctx, logger)
 
-	var errs []error
-	requeue, err := c.reconcile(ctx, workspace)
+	getMountObjectFunc := func(ctx context.Context, cluster logicalcluster.Path, ref tenancyv1alpha1.ObjectReference) (*unstructured.Unstructured, error) {
+		// TODO(sttts): do proper REST mapping.
+		resource := strings.ToLower(ref.Kind) + "s"
+		gvr := schema.GroupVersionResource{Resource: resource}
+		cs := strings.SplitN(ref.APIVersion, "/", 2)
+		if len(cs) == 2 {
+			gvr.Group = cs[0]
+			gvr.Version = cs[1]
+		} else {
+			gvr.Version = ref.APIVersion
+		}
+		return c.dynamicClusterClient.Cluster(cluster).Resource(gvr).Get(ctx, ref.Name, metav1.GetOptions{})
```

Comment:
It looks like the namespace of the reference isn't being used here. Is this intentional, or am I missing something?

Reply:
Idea is that mounts are cluster-scoped objects. Dont support namespace one for now

Reply:
Okay, got it. Thanks for the clarification.

Reply:
But good point. We need to document this.